uber/site_sections/registration.py

```diff
@@ -643,7 +643,11 @@ def new_checkin(self, session, id, badge_num='', ec_phone='', message='', group=
         elif existing:
             message = '{a.badge} already belongs to {a.full_name}'.format(a=existing[0])
         else:
-            badge_type, message = get_badge_type(badge_num)
+            message = check_range(badge_num, attendee.badge_type)
+            if not message:
+                maybe_dupe = session.query(Attendee).filter_by(badge_num=badge_num, badge_type=attendee.badge_type)
+                if maybe_dupe.count():
+                    message = 'That badge number already belongs to ' + maybe_dupe.first().full_name
```

Comment:
We already have a check above for badges which already belong to other people, so this whole ``if not message`` block is superfluous.